### PR TITLE
Fix lora grads for gemma4 vision

### PR DESCRIPTION
**Pull request by Datta0**
Transformers defines vision_tower after language_model for gemma4 causing issues with gradient hooks and lora not learning for vision layers. This fixes it.

### Gemma4: language model here comes before vision_model
```
class Gemma4Model(Gemma4PreTrainedModel):
    # we are filtering the logits/labels so we shouldn't divide the loss based on num_items_in_batch
    accepts_loss_kwargs = False

    def __init__(self, config: Gemma4Config):
        super().__init__(config)
        self.vocab_size = config.text_config.vocab_size

        language_model = AutoModel.from_config(config=config.text_config)
       
       # language model here comes before vision_model

        self.language_model = language_model
        self.vocab_size_per_layer_input = config.text_config.vocab_size_per_layer_input
        self.vision_tower = AutoModel.from_config(config.vision_config) if config.vision_config is not None else None
        self.embed_vision = Gemma4MultimodalEmbedder(config.vision_config, config.text_config) if config.vision_config is not None  else None
        self.audio_tower = AutoModel.from_config(config.audio_config) if config.audio_config is not None else None
        self.embed_audio = Gemma4MultimodalEmbedder(config.audio_config, config.text_config if config.audio_config is not None else None
        # Grab the ones from the child
        self._keys_to_ignore_on_load_unexpected = [
            f"language_model.{name}" for name in self.language_model._keys_to_ignore_on_load_unexpected
        ]
        self.post_init()
```

vs

### Gemma3: # language model here comes after vision_model
```
class Gemma3Model(Gemma3PreTrainedModel):
    # we are filtering the logits/labels so we shouldn't divide the loss based on num_items_in_batch
    accepts_loss_kwargs = False

    def __init__(self, config: Gemma3Config):
        super().__init__(config)
        self.vision_tower = AutoModel.from_config(config=config.vision_config)
        self.multi_modal_projector = Gemma3MultiModalProjector(config)
        self.vocab_size = config.text_config.vocab_size

        language_model = AutoModel.from_config(config=config.text_config)
        self.language_model = language_model
        self.post_init()
```

### CODE REVIEW

**Comment by Datta0** [comment]
Fixes: https://github.com/unslothai/unsloth/issues/5039
